doc/changelog/04-tactics/15071-namegen-no-glob.rst

```diff
@@ -0,0 +1,5 @@
+- **Changed:** Name generation does not avoid names of references
+  except for those from the current module and its ancestors (`#15071
+  <https://github.com/coq/coq/pull/15071>`_, fixes `#3523
+  <https://github.com/coq/coq/issues/3523>`_, by Gaëtan Gilbert).
+  The old behaviour may be restored using :flag:`Avoid Local Imported Names`.
```

Comment:
```suggestion
- **Changed:** Name generation no longer avoids names of references
  except for those from the current module and its ancestors (`#15071
  <https://github.com/coq/coq/pull/15071>`_, fixes `#3523
  <https://github.com/coq/coq/issues/3523>`_, by Gaëtan Gilbert).
  The old behavior can be restored using :flag:`Avoid Local Imported Names`.
```

"Avoid Local Imported Names" is a bit of a mouthful but I have no better suggestion.